providers/ProfileProvider.tsx

```diff
@@ -48,17 +59,11 @@ const ProfileProvider = ({ children }: { children: ReactNode }) => {
       }
     };
 
-    if (!username) {
-      setProfile(null);
-      setFollowers([]);
-      setSongs([]);
-      setTopSongMetadata(null);
-      return;
-    }
-    init();
+    init().then(() => setLoading(false));
```

Comment:
i like this usage of `.then` 🧠 for setting of loading state.
- in the case of `failure` or `error`, will we be stuck in `loading` state? aka, do we need the `finally` hook?